cpp/daal/include/sycl/internal/buffer_utils.h

```diff
@@ -129,10 +129,16 @@ class BufferConverterTo
     {
         _st = services::Status();
 
-        auto buffer    = _src.template get<DataType>();
-        auto subbuffer = buffer.getSubBuffer(_offset, _size);
-
-        _dest = subbuffer;
+        auto buffer = _src.template get<DataType>();
+        if ((_offset == 0) && (_size == buffer.size()))
```

Comment:
Why this dispatching is done here, but not [there](https://github.com/oneapi-src/oneDAL/blob/c19d88eca7c627e4cb79833062c695bdddfbf986/cpp/daal/include/services/buffer.h#L190)?